### ISSUE TITLE
Different Profiles for different settings [enhancement]

### ISSUE THREAD

**Issue by ant4n** [noah-nuebling/Mac-Mouse-Fix#859]
## Description
In the status bar a drop down menu with created profiles should be displayed.
If you are using a specific program with various unique shortcuts for example, a profile custom named to indicate the intended software will load different shortcuts / settings into the mouse buttons.

## Rationale
I like having global shortcuts in my mouse to navigate the different pages on my mac, however I also use photoshop and music software, which has a lot of complex shortcuts as well as external peripherals. If I want to play the keyboard while changing settings quickly, having one hand inputting complex shortcuts through the mouse would be convenient. 

At the moment though, having global shortcuts in a program that doesnt use them is pointless. and having program specific shortcuts when not using the program is pointless as well becuase they may not affect the computer globally.

I think a drop down menu to choose between profiles custom made for photoshop, Ableton, premiere pro would be efficient. 

I wouldnt necessarily want settings that automatically come on when using a certain program if I do choose to use global settings (like swapping between screens instead of navigating Ableton)

## Additional info
–

**Comment by noah-nuebling** [comment]
Hi there @ant4n, thanks for your feedback!

I'm working on implementing something like this. 

I thought about doing profiles very similar to how you describe, but I didn't really see the benefit over having app-specific settings.

So it's very interesting to me that you say you'd prefer profiles over app-specific settings. It would be very interesting if you could elaborate on that that so I can understand you use case better? 

Specifically this:

> I wouldnt necessarily want settings that automatically come on when using a certain program if I do choose to use global settings (like swapping between screens instead of navigating Ableton)

**Comment by ant4n** [comment]
Yes, for example I like using the side buttons on my mouse to select
various instrument tracks on Ableton. But if I want to move to another
screen (what apple calls 'spaces') on mac I would have to reach to my
keyboard or my macbook to do it. Whereas I could instead go to the status
bar, select the global profile I hypothetically set in advance, and use the
side buttons to move to another space.

This is my personal example but it would be useful if I switch to safari
and use the same side buttons to add or remove tabs. Or go to spotify click
both buttons to search, use up button to add it to my playlist.

Essentially have infinite shortcuts on the mouse.

If you didn't want to go to status bar you could even cycle through
profiles with a unique button shortcut on your mouse.

**Comment by noah-nuebling** [comment]
Hmm interesting, thanks for getting back!

How about if the mappings on your mouse would be dependent on the app under your mouse pointer? So e.g. if you're working in Ableton in fullscreen, you could still just move your pointer over the status bar to use your default profile which lets you switch spaces.

Would you still prefer having profiles in that case?

The idea of having muliple profiles to cycle through at the press of a button is also very interesting and would be extremely powerful, but for me personally I don't think I would use it because I worry it might be confusing, hard to set up, and not provide a real benefit. 

I don't think the benefit would really be there for me since you can already put a lot of actions on clicking, double clicking, clicking and dragging, etc. Especially if you could assign custom actions to click and drag, (which I do plan, but I'm not sure when it's coming), you could easily set 12 different actions to a single button just by clicking, double clicking or triple clicking and then dragging up, down, left, or right.

I just think that would be more than enough custom actions for my personal pro workflows, and I think it would be much easier to setup and use than having profiles you can cycle through. I'd also definitely forget all the time which profile I'm currently on. I guess you could show it in the status bar, but I just don't imagine the experience to be great for me personally.

But everyones different, especially when it comes to how we use computers! So I'd be very interested to hear your perspective.

**Comment by john1625b** [comment]
this is really useful for gaming. each game needs a different profile since the hotkeys are different for each game. for instance i need 1, 2 and 3 mapped to different buttons on the mouse for starcraft
